libs/shared/ui/src/lib/components/table/table.tsx

```diff
@@ -34,16 +39,43 @@ export function Table(props: TableProps) {
     defaultData,
     filterData,
     setFilterData,
+    autoscrollToNew = false,
   } = props
 
   const ALL = 'ALL'
   const [currentFilter, setCurrentFilter] = useState(ALL)
+  const [previousIds, setPreviousIds] = useState(JSON.stringify(defaultData?.map(({ id }) => id)))
+
+  const [autoScrollTimer] = useState<number>(new Date().getTime())
```

Comment:
Why isn't a simple constant ? you don't use setState here

Reply:
Because I want to initialize this value when we first render the table to get the moment in time it has been initialized. It allows me to compute the time spent since creation. And it allows me to block the autoscroll feature for the first second to avoid a glitch

Reply:
ok, I see, after checking on the web, it isn't the best way to do

You can use ```  const ref = useRef();```
https://stackoverflow.com/questions/65027884/which-is-the-right-way-to-detect-first-render-in-a-react-component